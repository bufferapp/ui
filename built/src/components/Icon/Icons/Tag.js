import React from 'react';
import createIconComponent from '../utils/createIconComponent';
const TagIcon = createIconComponent({
    content: (React.createElement("g", null,
        React.createElement("path", { fillRule: "evenodd", d: "M14 1H8.414a1 1 0 00-.707.293l-7 7a1 1 0 000 1.414l5.586 5.586a1 1 0 001.414 0l7-7A1 1 0 0015 7.586V2a1 1 0 00-1-1zm-2.5 5a1.5 1.5 0 100-3 1.5 1.5 0 000 3z", clipRule: "evenodd" }))),
});
TagIcon.displayName = 'TagIcon';
export default TagIcon;
//# sourceMappingURL=Tag.js.map