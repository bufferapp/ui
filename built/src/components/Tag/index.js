export { default } from './Tag';
//# sourceMappingURL=index.js.map