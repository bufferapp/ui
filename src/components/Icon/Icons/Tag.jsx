/** WARNING
 * This file is automatically generated by the `npm run gen:icons` command.
 * ALL changes will be overwritten.
 */
import React from 'react';
import createIconComponent from '../utils/createIconComponent';

const TagIcon = createIconComponent({
  content: (
    <g>
      <path
        fillRule="evenodd"
        d="M14 1H8.414a1 1 0 00-.707.293l-7 7a1 1 0 000 1.414l5.586 5.586a1 1 0 001.414 0l7-7A1 1 0 0015 7.586V2a1 1 0 00-1-1zm-2.5 5a1.5 1.5 0 100-3 1.5 1.5 0 000 3z"
        clipRule="evenodd"
      />
    </g>
  ),
});
TagIcon.displayName = 'TagIcon';

export default TagIcon;
