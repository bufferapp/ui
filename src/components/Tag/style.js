import styled from 'styled-components';
import { fontSize, fontWeightMedium, lineHeight } from '../style/fonts';
import textColors, * as Colors from '../style/colors';

export const Tag = styled.span`
  display: inline-block;
  padding: 4px 8px;
  font-weight: ${fontWeightMedium};
  font-size: ${fontSize};
  line-height: ${lineHeight};
  margin-left: 8px;
  border-radius: 100px;
  color: ${(props) => textColors[props.textColor]};
  background-color: ${(props) => Colors[props.color]};

  &:hover {
    color: ${(props) => textColors[props.textColor]};
  }
`;
