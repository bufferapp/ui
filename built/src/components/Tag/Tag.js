var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import React from 'react';
import PropTypes from 'prop-types';
import * as Styles from './style';
const Tag = (_a) => {
    var { children } = _a, props = __rest(_a, ["children"]);
    return (React.createElement(Styles.Tag, Object.assign({}, props), children));
};
Tag.propTypes = {
    children: PropTypes.node.isRequired,
    color: PropTypes.string,
    textColor: PropTypes.string,
};
Tag.defaultProps = {
    color: 'green',
    textColor: 'white',
};
export default Tag;
//# sourceMappingURL=Tag.js.map