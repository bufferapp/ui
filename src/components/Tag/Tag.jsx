import React from 'react';
import PropTypes from 'prop-types';
import * as Styles from './style';

const Tag = ({ children, ...props }) => (
  <Styles.Tag {...props}>{children}</Styles.Tag>
);

Tag.propTypes = {
  /** The actual content of the tag. */
  children: PropTypes.node.isRequired,
  /** The color of the tag. */
  color: PropTypes.string,
  /** The color of the text. It can be: `white`, `gray`, `grayDark`, `blue` and `red`. */
  textColor: PropTypes.string,
};

Tag.defaultProps = {
  color: 'green',
  textColor: 'white',
};

export default Tag;
