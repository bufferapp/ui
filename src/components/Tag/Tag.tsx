import React from 'react'
import PropTypes from 'prop-types'
import * as Styles from './style'

// @ts-expect-error TS(7031) FIXME: Binding element 'children' implicitly has an 'any'... Remove this comment to see the full error message
const Tag = ({ children, ...props }) => (
  <Styles.Tag {...props}>{children}</Styles.Tag>
)

Tag.propTypes = {
  /** The actual content of the tag. */
  children: PropTypes.node.isRequired,
  /** The color of the tag. */
  color: PropTypes.string,
  /** The color of the text. It can be: `white`, `gray`, `grayDark`, `blue` and `red`. */
  textColor: PropTypes.string,
}

Tag.defaultProps = {
  color: 'green',
  textColor: 'white',
}

export default Tag
