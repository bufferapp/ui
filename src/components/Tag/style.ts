import styled from 'styled-components'
import { fontSizeSmall, fontWeightMedium, lineHeight } from '../style/fonts'
import textColors, * as Colors from '../style/colors'

export const Tag = styled.span`
  display: flex;
  padding: 4px 8px;
  font-weight: ${fontWeightMedium};
  font-size: ${fontSizeSmall};
  line-height: ${lineHeight};
  border-radius: 100px;
  color: ${(props) =>
    // @ts-expect-error TS(7053) FIXME: Element implicitly has an 'any' type because expre... Remove this comment to see the full error message
    textColors[props.textColor]};
  background-color: ${(props) =>
    // @ts-expect-error TS(7053) FIXME: Element implicitly has an 'any' type because expre... Remove this comment to see the full error message
    Colors[props.color]};

  &:hover {
    color: ${(props) =>
      // @ts-expect-error TS(7053) FIXME: Element implicitly has an 'any' type because expre... Remove this comment to see the full error message
      textColors[props.textColor]};
  }
`
