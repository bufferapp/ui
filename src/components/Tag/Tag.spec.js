/* eslint-disable react/jsx-filename-extension */
import snap from 'jest-auto-snapshots';
import 'jest-styled-components';

import Tag from './Tag';

snap(Tag, './Tag.jsx');