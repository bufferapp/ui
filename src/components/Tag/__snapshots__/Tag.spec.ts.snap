// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-auto-snapshots > Tag Matches snapshot when passed all props 1`] = `
<styled.span
  color="jest-auto-snapshots String Fixture"
  textColor="jest-auto-snapshots String Fixture"
>
  <NodeFixture />
</styled.span>
`;

exports[`jest-auto-snapshots > Tag Matches snapshot when passed only required props 1`] = `
<styled.span
  color="green"
  textColor="white"
>
  <NodeFixture />
</styled.span>
`;
